### ISSUE TITLE
Config Enhancement & UI Improve & Function Improve & BUGs

### ISSUE THREAD

**Issue by SnowCutieOwO** [FrostbyteGames1/Frostbytes-Improved-Inventory#18]
* Add `DoPlaySoundsOnZoom` option:
    * Determines if you use the zoom function, the sound will play or not
* Add `ZoomMultiplier` option:
    * The existing setting seems different from the mod `logical-zoom` and optifine has.

* Moveable UI:
    * The function of Slot Cycle will cover the InventoryHUD+ UI, please resolve this conflict or add a moveable UI like JourneyMap does to the potion effects.

* Visible Item Refill
    * Show the remaining items that can be refilled on the position of actionbar.

* BUG:
    * Sort function sometimes will left a slot of item on mouse pointer.
    * Idk why, but the shulker box not being rendered. Is there any shortcut key or it will automatically show?
    * Zoom function would be disabled until you open the *esc* menu.

<3

**Comment by FrostbyteGames1** [comment]
As of v1.9.4:

> Add `DoPlaySoundsOnZoom` option:

There is now an option to disable the zoom sound.

> Add `ZoomMultiplier` option:

Zoom can now be adjusted by scrolling up or down while the zoom key is held.

> Moveable UI:

All screen elements now have config options to adjust their positions.

> Visible Item Refill

There is now a config option to add the number of matching items in the inventory is to the icon of the held itemstack

> Sort function sometimes will left a slot of item on mouse pointer.

This should now be resolved.

> Idk why, but the shulker box not being rendered. Is there any shortcut key or it will automatically show?

As long as the config option is enabled and the shulker box is not empty, the preview should automatically show.

> Zoom function would be disabled until you open the _esc_ menu.

I'm not sure what is meant by this? The zoom function only works when in the world and no containers or screens are open. If the zoom key is held and a container or screen is opened, the FOV should return to normal.

**Comment by SnowCutieOwO** [comment]
> As of v1.9.4:
> 
> > Add `DoPlaySoundsOnZoom` option:
> 
> There is now an option to disable the zoom sound.
> 
> > Add `ZoomMultiplier` option:
> 
> Zoom can now be adjusted by scrolling up or down while the zoom key is held.
> 
> > Moveable UI:
> 
> All screen elements now have config options to adjust their positions.
> 
> > Visible Item Refill
> 
> There is now a config option to add the number of matching items in the inventory is to the icon of the held itemstack
> 
> > Sort function sometimes will left a slot of item on mouse pointer.
> 
> This should now be resolved.
> 
> > Idk why, but the shulker box not being rendered. Is there any shortcut key or it will automatically show?
> 
> As long as the config option is enabled and the shulker box is not empty, the preview should automatically show.
> 
> > Zoom function would be disabled until you open the _esc_ menu.
> 
> I'm not sure what is meant by this? The zoom function only works when in the world and no containers or screens are open. If the zoom key is held and a container or screen is opened, the FOV should return to normal.

Woah! Thanks very much adding those suggestions.

Um, about the last one, I mean that:

1. Open the game.
2. Play in a single player or multi player world, and you could found that the zoom function not working.
3. Open the pause menu, and close it, the game will froze for a vert short amount of time, and the zoom function could be used.

The truth is, every time I open my game, I should suffer from such a situation and I'm troubled by it.

**Comment by FrostbyteGames1** [comment]
I haven't been able to replicate the zoom behavior that you are describing. Do you have any other mods installed that might be interacting with this one?

**Comment by SnowCutieOwO** [comment]
> I haven't been able to replicate the zoom behavior that you are describing. Do you have any other mods installed that might be interacting with this one?

I do installed `logical-zoom`, but since I installed your mod, I disabled that manually.
And I've testes again, if you entered game and does not open the pause menu, the sort function would also unavaliable :/

**Comment by FrostbyteGames1** [comment]
Are there any conflicting keybinds?

**Comment by SnowCutieOwO** [comment]
> Are there any conflicting keybinds?

No, I believe that this is a bug, since there's no any other mod could apply such a problem :/

**Comment by SnowCutieOwO** [comment]
> > Are there any conflicting keybinds?
> 
> No, I believe that this is a bug, since there's no any other mod could have such a problem :/

**Comment by FrostbyteGames1** [comment]
It will take some time to figure out the cause of this, but in the meantime, I have released v1.9.6, which adds the container search feature.

**Comment by SnowCutieOwO** [comment]
> It will take some time to figure out the cause of this, but in the meantime, I have released v1.9.6, which adds the container search feature.

Have you added 1.20.6 support?

**Comment by SnowCutieOwO** [comment]
> Yes, v1.8.5

Is there will be version 1.9 or higher for 1.20.6? I really hope I could use what I suggested on 1.20.6 :)

**Comment by FrostbyteGames1** [comment]
v1.8.5 is a backport of v1.9.6, it has all of the same features

**Comment by SnowCutieOwO** [comment]
> v1.8.5 is a backport of v1.9.6, it has all of the same features

Got it. I'll have a try. :)

**Comment by SnowCutieOwO** [comment]
Unfortunately when I used the newer version, the game seems really strange.
At first I went into a server and found all blocks and entities are invisible, second time I tried open Minecraft with only `yetanotherconfig`, your mod and `fabric-api`, the game became dull and will frozen on enter any world or server.
I have three logs here, if you want, this is the link:

https://bytebin.lucko.me/on8QsilqEl
*I've checked content here but I cannot figure out anything. :(*

https://bytebin.lucko.me/H2qlWPKG8A
(With latest version for 1.20.6, added `yetanotherconfiglib`, `fabric-api`)

https://bytebin.lucko.me/4oeAyTZnlC
(With 1.8.4 for 1.20.6, added `yetanotherconfiglib`, `fabric-api`)

And btw, what I suggested is the floating search bar which does not occupy any inventory. :/

However thanks for making these, they're still cool :)

**Comment by FrostbyteGames1** [comment]
The rendering bug for the 1.20.6 version should now be fixed by v1.8.6

**Comment by SnowCutieOwO** [comment]
Sorry bother again, could you please add a blacklist or whitelist for auto-refill?

**Comment by FrostbyteGames1** [comment]
Is there a specific reason you are wanting a blacklist/whitelist? I could probably add one, but the way I go about implementing it would depend on the primary use case.

**Comment by SnowCutieOwO** [comment]
> Is there a specific reason you are wanting a blacklist/whitelist? I could probably add one, but the way I go about implementing it would depend on the primary use case.

If I want use lava bucket for multiple times, I would need this function. When I was using lava or water bucket, normal refill mod will replace the bucket into empty one, so I had to open my inventory and drag them to my hotbar again. Very annoying, and I just want mod does not affect them.
Another example is the shulker box. If I had two shulker box, one of them is full. When I tried to put the empty one down, the other one will appear on my hand which is not intended.
